visual-js/visual-snapshots/src/app/worker/pdf-page-snapshot-uploader.ts

```diff
@@ -14,23 +16,24 @@ export class PdfPageSnapshotUploader {
     buildId: string,
     pdfFilePath: string,
     pageNumber: number,
-    suiteName: string | undefined,
+    suiteNameFormat: string | undefined,
     testNameFormat: string | undefined,
     snapshotNameFormat: string | undefined
   ) {
     const pdfFile = await this.pdfFileLoader.loadPdfFile(pdfFilePath);
     const page = await pdfFile.getPage(pageNumber);
 
-    const filename = path.basename(pdfFile.path);
+    const pdfFileMetadata = buildFileMetadata(pdfFilePath, pageNumber);
     const testName = testNameFormat
-      ? formatString(testNameFormat, { filename })
+      ? formatStringWithFileMetadata(testNameFormat, pdfFileMetadata)
```

Comment:
Do we want to enable all templates for each snapshot, test, and suite name? I think this way we make it more flexible but I am fine with limiting it

Reply:
If providing in that context makes sense I think the more flexibility the better provided we have pretty good defaults so it doesn't require someone to look through all the docs. I know that part is a separate ticket though.